### ISSUE TITLE
[Bug] Weird loss when training form scratch for RWKV7

### ISSUE THREAD

**Issue by necrophagists** [fla-org/flash-linear-attention#218]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

<img width="197" alt="Image" src="https://github.com/user-attachments/assets/3b4fc864-dfd0-4cb9-9ef3-e739ed827991" />

The datasets：10% of pile.
Model size: 0.6b. Tokenizer: custom tokenizer.
I did not use the official RWKV-7 weights, but instead initialized a model using the config and then trained it.
In the figure, there are three loss curves. The purple one is the model initialized with the default parameters of FLA style. The green one is the model with the initialization method  of RWKV-LM. The black one is a 0.6b model with the QWen-2 architecture.
These loss curves  of rwkv are very abnormal.
Then I tried to train on  small webtext data that all sequences length is 1024 （no need left pad. The loss is still so weird.

### Steps to Reproduce the Bug

find one datasets

```
from fla.models.rwkv7 import RWKV7ForCausalLM,RWKV7Config
...    
config = RWKV7Config.from_pretrained(args.config_path)
model = RWKV7ForCausalLM(
            config
        ). #get origin model
...
```
Then train.

### Expected Behavior

The loss should converge normally. 

### Environment Information

1. Torch:
2. Triton:

**Comment by necrophagists** [comment]
@

> Could you show me your training configs more elaborately

environment: triton 3.1.0 transformers 4.49.0  torch 2.5.1 

the model config.
```
{
  "_attn_implementation_autoset": true,
  "a_low_rank_dim": 96,
  "architectures": [
    "RWKV7ForCausalLM"
  ],
  "attn": null,
  "attn_mode": "chunk",
  "auto_map": {
    "AutoConfig": "modeling_rwkv7.RWKV7Config",
    "AutoModel": "modeling_rwkv7.RWKV7Model",
    "AutoModelForCausalLM": "modeling_rwkv7.RWKV7ForCausalLM"
  },
  "pad_token_id": 0,
  "bos_token_id": 1,
  "decay_low_rank_dim": 96,
  "eos_token_id": 2,
  "fuse_cross_entropy": true,
  "fuse_norm": true,
  "gate_low_rank_dim": 320,
  "head_dim": 64,
  "hidden_act": "sqrelu",
  "hidden_ratio": 4.0,
  "hidden_size": 1024,
  "initializer_range": 0.02,
  "intermediate_size": 4096,
  "max_position_embeddings": 2048,
  "model_type": "rwkv7",
  "norm_bias": true,
  "norm_eps": 1e-05,
  "norm_first": true,
  "num_heads": null,
  "num_hidden_layers": 30,
  "tie_word_embeddings": false,
  "torch_dtype": "float32",
  "transformers_version": "4.48.2",
  "use_cache": true,
  "v_low_rank_dim": 64,
  "vocab_size": 127408
}
```



the training config:
```
  --deepspeed ${ds_config_file} \
  --config_path ${config_path} \
  --token_model_file ${token_model_file} \
  --load_from_pretrain False \
  --model_path ${model_path} \
  --data_path ${data_path} \
  --model_max_length ${cutoff_len} \
  --gradient_checkpointing False \
  --resume_from_checkpoint False \
  --per_device_train_batch_size 4 \
  --per_device_eval_batch_size 1 \
  --gradient_accumulation_steps 4  \
  --dataloader_num_workers 16 \
  --num_train_epochs 1 \
  --save_strategy "epoch" \
  --do_train True \
  --do_eval False \
  --evaluation_strategy "no" \
  --save_total_limit 1 \
  --learning_rate 8e-4 \
  --adam_beta1 0.9 \
  --adam_beta2 0.95 \
  --adam_epsilon 1e-8 \
  --weight_decay 0.01 \
  --warmup_ratio 0.01 \
  --lr_scheduler_type "constant" \
  --logging_steps 5 \
  --bf16 True \
  --seed 1119 \
  --report_to "tensorboard" \
```

ds config
```
{
  "optimizer": {
    "type": "AdamW",
    "params": {
      "lr": "auto",
      "betas": "auto",
      "eps": "auto",
      "weight_decay": "auto"
    }
  },
  "fp16": {
    "enabled": false,
    "loss_scale": 0,
    "loss_scale_window": 1000,
    "initial_scale_power": 16,
    "hysteresis": 2,
    "min_loss_scale": 1
  },
  "bf16": {
    "enabled": true
  },
  "scheduler": {
    "type": "WarmupLR",
    "params": {
      "warmup_min_lr": "auto",
      "warmup_max_lr": "auto",
      "warmup_num_steps": "auto",
      "warmup_type": "log"
    }
  },
  "zero_optimization": {
    "stage": 1,
    "allgather_partitions": true,
    "allgather_bucket_size": 1e9,
    "overlap_comm": false,
    "reduce_scatter": true,
    "reduce_bucket_size": 1e9,
    "contiguous_gradients": true,
    "round_robin_gradients": true
  },
  "gradient_accumulation_steps": "auto",
  "gradient_clipping": "auto",
  "steps_per_print": 2000,
  "train_batch_size": "auto",
  "train_micro_batch_size_per_gpu": "auto"
```

**Comment by yzhangcs** [comment]
@necrophagists Looks normal for me, could you try it by https://github.com/fla-org/flame?

![Image](https://github.com/user-attachments/assets/88539f77-9628-410a-b598-1f87682ae3e1)

**Comment by necrophagists** [comment]
Certainly! I would be happy to give it a try. Could you  share the configuration for this experiment? Additionally, I was wondering if your experiment was conducted on H-series or A-series GPUs, as I believe this might have an impact. Lastly, I would like to request if you could attempt to reproduce my issue within the Transformers training framework,I have also noticed that when loading the 2.9B models for incremental training, the loss remains at a relatively high level (around 10). Thank you in advance for your assistance!

**Comment by necrophagists** [comment]
thank you. I will check it carefully.

**Comment by necrophagists** [comment]
@yzhangcs Hi,can you offer you train.toml？Thank you！

**Comment by yzhangcs** [comment]
@necrophagists 
https://github.com/fla-org/flame/blob/main/train.toml
